hadoop-hdds/rocks-native/pom.xml

```diff
@@ -0,0 +1,405 @@
+                                </goals>
+                            </execution>
+                            <execution>
+                                <id>build-snappy</id>
+                                <phase>process-sources</phase>
+                                <configuration>
+                                    <tasks>
+                                        <mkdir dir="${project.build.directory}/snappy/lib"/>
+                                        <exec executable="cmake" failonerror="true" dir="${project.build.directory}/snappy/lib">
+                                            <arg line="${project.build.directory}/snappy/snappy-${snappy.version}"/>
+                                            <env key="CFLAGS" value="-fPIC"/>
+                                            <env key="CXXFLAGS" value="-fPIC"/>
+                                        </exec>
+                                        <exec executable="make" dir="${project.build.directory}/snappy/lib" failonerror="true"/>
+                                    </tasks>
+                                </configuration>
+                                <goals>
+                                    <goal>run</goal>
+                                </goals>
+                            </execution>
+                            <execution>
+                                <id>build-rocksjava</id>
+                                <phase>process-resources</phase>
+                                <configuration>
+                                    <tasks>
+                                        <exec executable="chmod" failonerror="true">
+                                            <arg line="-R"/>
+                                            <arg line="775"/>
+                                            <arg line="${project.build.directory}/rocksdb/rocksdb-${rocksdb.version}"/>
+                                        </exec>
+                                        <exec executable="make" dir="${project.build.directory}/rocksdb/rocksdb-${rocksdb.version}" failonerror="true">
+                                            <arg line="EXTRA_CXXFLAGS='-fPIC -I${project.build.directory}/snappy/lib -I${project.build.directory}/snappy/snappy-${snappy.version} -I${project.build.directory}/lz4/lz4-${lz4.version}/lib -I${project.build.directory}/zstd/zstd-${zstd.version}/lib -I${project.build.directory}/zstd/zstd-${zstd.version}/lib/dictBuilder -I${project.build.directory}/bzip2/bzip2-${bzip2.version} -I${project.build.directory}/zlib/zlib-${zlib.version}'"/>
+                                            <arg line="EXTRA_LDFLAGS='-L${project.build.directory}/snappy/lib -L${project.build.directory}/lz4/lz4-${lz4.version}/lib -L${project.build.directory}/zstd/zstd-${zstd.version}/lib -L${project.build.directory}/bzip2/bzip2-${bzip2.version} -L${project.build.directory}/zlib/zlib-${zlib.version}'"/>
+                                            <arg line="-j4"/>
```

Comment:
Is this the maximum the Github worker can run at with 7GB RAM? Does `-j6` fail the build?

Reply:
TODO: Use `nproc` or equivalent

Reply:
done